src/main/java/com/aws/greengrass/mqtt/bridge/model/MqttMessage.java

```diff
@@ -5,22 +5,55 @@
 
 package com.aws.greengrass.mqtt.bridge.model;
 
+import com.aws.greengrass.mqttclient.v5.Publish;
+import com.aws.greengrass.mqttclient.v5.UserProperty;
+import lombok.Builder;
 import lombok.NonNull;
 import lombok.Value;
 
-// TODO support mqtt5
+import java.util.Collections;
+import java.util.List;
+
 @Value
+@Builder
```

Comment:
Use `@Builder(toBuilder = true)` so you can just do `return this.toBuilder().topic(topic).build()` to modify the topic

Reply:
oh nice!